server/src/main/java/com/sauron/service/CurrentBalanceServiceImpl.java

```diff
@@ -1,14 +1,33 @@
 package com.sauron.service;
 
+import com.sauron.model.Transaction;
+import com.sauron.model.entities.TransactionDirection;
 import org.springframework.stereotype.Service;
 
 import java.math.BigDecimal;
+import java.util.Collection;
 
 @Service
 public class CurrentBalanceServiceImpl implements CurrentBalanceService {
 
+    private final TransactionService transactionService;
+
+    public CurrentBalanceServiceImpl(TransactionService transactionService) {
+        this.transactionService = transactionService;
+    }
+
     @Override
     public BigDecimal getCurrentBalance() {
-        return new BigDecimal("123.45");
+        Collection<Transaction> transactions = transactionService.getAllTransactions();
+
+        return transactions.stream().map(this::convertToAmount).reduce(BigDecimal.ZERO, BigDecimal::add);
```

Comment:
I guess it's temporary. The transactions fetched from the bank API (for some slice of time only) does not necessarily sum up to the current balance.
If it's temporary, please add some TODO because it may be misleading.

Reply:
For now this is using method that is taking ALL transactions ;) obviously implementation of this method depends of API which will be provided by other banks so probably it will be a good thing to discuss about some abstractions for this.